src/main/java/elberger/maze/Maze.java

```diff
@@ -4,145 +4,152 @@
-			neighbors.push(eastNeighbor);
-			neighbors.push(southNeighbor);
+			neighbors.add(maze[currentCell.getX()][currentCell.getY() + 1]);
 		}
-
-		else if (cell.getX() == x)
+		//get west neighbor
+		if (currentCell.getX() != 0)
 		{
-			Cell northNeighbor = new Cell(cell.getX() - 1, cell.getY());
-			Cell eastNeighbor = new Cell(cell.getX(), cell.getY() - 1);
-			Cell westNeighbor = new Cell(cell.getX(), cell.getY() + 1);
-
-			neighbors.push(northNeighbor);
-			neighbors.push(eastNeighbor);
-			neighbors.push(westNeighbor);
+			neighbors.add(maze[currentCell.getX() - 1][currentCell.getY()]);
 		}
-
-		else
+		//get east neighbor
+		if (currentCell.getX() < (getWidth() - 1))
 		{
-			Cell northNeighbor = new Cell(cell.getX() - 1, cell.getY());
-			Cell eastNeighbor = new Cell(cell.getX(), cell.getY() - 1);
-			Cell westNeighbor = new Cell(cell.getX(), cell.getY() + 1);
-			Cell southNeighbor = new Cell(cell.getX() + 1, cell.getY());
-
-			neighbors.push(northNeighbor);
-			neighbors.push(eastNeighbor);
-			neighbors.push(westNeighbor);
-			neighbors.push(southNeighbor);
+			neighbors.add(maze[currentCell.getX() + 1][currentCell.getY()]);
 		}
 
 		Collections.shuffle(neighbors);
 		return neighbors;
 	}
 
-	public void setNeighbors(LinkedList<Cell> neighbors)
-	{
-		this.neighbors = neighbors;
-	}
-
-
-	public Cell startingCell()
+	private void breakWalls(Cell currentCell, Cell nextCell)
 	{
-		Random x = new Random();
-		Random y = new Random();
-		Cell cell = new Cell(x.nextInt(getX()), y.nextInt(getY()));
-		return cell;
+		//neighbor = north
+		if (currentCell.getY() == nextCell.getY() + 1)
+		{
+			currentCell.setNorthWall(false);
+			nextCell.setSouthWall(false);
+		}
+		//neighbor = south
+		else if (currentCell.getY() == nextCell.getY() - 1)
+		{
+			currentCell.setSouthWall(false);
+			nextCell.setNorthWall(false);
+		}
+		//neighbor = east
+		else if (currentCell.getX() == nextCell.getX() + 1)
+		{
+			currentCell.setEastWall(false);
+			nextCell.setWestWall(false);
+		}
+		//neighbor = west
+		else if (currentCell.getX() == nextCell.getX() - 1)
+		{
+			currentCell.setWestWall(false);
+			nextCell.setEastWall(false);
+		}
 	}
 
 	public void searchPath(Cell cell)
 	{
+		Stack<Cell> stack = new Stack<>();
 		stack.push(cell);
-		visited.push(cell);
+		cell.setVisited(true);
 
 		while (!stack.isEmpty())
 		{
-			Cell currentCell = (Cell) stack.peek();
-			Cell nextCell = getNeighbors(currentCell).getFirst();
-			if (!visited.contains(nextCell))
+			Cell currentCell = stack.pop();
+			Cell nextCell = getNeighbors(currentCell).get(0);
+
+			if (!nextCell.isVisited())
 			{
-				visited.push(nextCell);
-				maze[nextCell.getX()][nextCell.getY()] = ".";
-				stack.push(nextCell);
+				nextCell.setVisited(true);
+				breakWalls(currentCell, nextCell);
+				stack.addAll(getNeighbors(currentCell));
 			}
 			else
 			{
-				stack.pop();
-				searchPath(nextCell);
+				for (int i = 0; i < getNeighbors(currentCell).size(); i++)
```

Comment:
You are calling `getNeighbors()` here 3 times.
What you've forgotten is that getNeighbors() returns a List that is SHUFFLED and be different each time it's called.